docs/overview.rst

```diff
@@ -515,6 +515,36 @@ without extra cost because the result is cached by the system.
 
     assert service.types is types
 
+.. _annotations:
+
+Customizing behavior with annotations
+-------------------------------------
+
+Thrift supports adding annotations to defined types, fields, references to
+primitive types, etc. These may be used to customize behavior of ``thriftrw``.
+
+The following annotations are supported by ``thriftrw``.
+
+``py.primitive_type``
```

Comment:
Yeah, I misunderstood. to_primitive is nearly to_json. I don’t recommend altering the language-level representation. You’ve made the right choices for Python. JavaScript is special since it doesn’t seem uncanny to duck-type everything.